core/ops.rs

```diff
@@ -27,86 +27,115 @@ use std::task::Poll;
 use v8::fast_api::CFunctionInfo;
 use v8::fast_api::CTypeInfo;
 
+pub type RealmIdx = usize;
+pub type PromiseId = i32;
+pub type OpId = usize;
+
 /// Wrapper around a Future, which causes that Future to be polled immediately.
 ///
 /// Background: ops are stored in a `FuturesUnordered` structure which polls
 /// them, but without the `OpCall` wrapper this doesn't happen until the next
 /// turn of the event loop, which is too late for certain ops.
-pub struct OpCall<T>(MaybeDone<Pin<Box<dyn Future<Output = T>>>>);
+pub struct OpCall {
+  realm_idx: RealmIdx,
+  promise_id: PromiseId,
+  op_id: OpId,
+  fut: MaybeDone<Pin<Box<dyn Future<Output = OpResult>>>>,
```

Comment:
I'm thinking that maybe we should store `Result<R, Error>` where `R: Serialize + 'static`. `OpResult` feels like an unnecessary indirection in this case.

Reply:
It would also allow us to remove `.map()` when queueing the ops

Reply:
@matt could you try to address that?

Reply:
I'll take a look at this, but if we're storing generic types it's either going to be `Box` or enum. In this case, `OpResult` is actually a good idea I think.